src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Base64Validator.cs

```diff
@@ -0,0 +1,201 @@
+            {
+                while (indexOfPaddingInvalidOrWhitespace >= 0)
+                {
+                    byte byteToValidate = base64TextUtf8[indexOfPaddingInvalidOrWhitespace];
+                    if (IsWhitespace(byteToValidate))
+                    {
+                        // Bytes to be ignored (e,g, whitespace...) should not count towards the length.
+                        length--;
+                    }
+                    else if (byteToValidate == EncodingPad)
+                    {
+                        // There can be at most 2 padding chars.
+                        if (paddingCount == 2)
+                        {
+                            decodedLength = 0;
+                            return false;
+                        }
+
+                        paddingCount++;
+                    }
+                    else
+                    {
+                        // An invalid char was encountered.
+                        decodedLength = 0;
+                        return false;
+                    }
+
+                    if (indexOfPaddingInvalidOrWhitespace == base64TextUtf8.Length - 1)
+                    {
+                        // The end of the input has been reached.
+                        break;
+                    }
+
+                    // If no padding is found, slice and use IndexOfAnyExcept to look for the next invalid char.
+                    if (paddingCount == 0)
+                    {
+                        indexOfPaddingInvalidOrWhitespace = base64TextUtf8
+                            .Slice(indexOfPaddingInvalidOrWhitespace + 1, base64TextUtf8.Length - indexOfPaddingInvalidOrWhitespace - 1)
+                            .IndexOfAnyExcept(validBase64Bytes)
+                            + indexOfPaddingInvalidOrWhitespace + 1; // Add current index offset.
+                    }
+                    // If padding is already found, simply increment, as the common case might have 2 sequential padding chars.
+                    else
+                    {
+                        indexOfPaddingInvalidOrWhitespace++;
+                    }
+                }
+
+                // If the invalid chars all consisted of whitespace, the input will be empty.
+                if (length == 0)
+                {
+                    decodedLength = 0;
+                    return true;
+                }
+            }
+
+            if (length % 4 != 0)
+            {
+                decodedLength = 0;
+                return false;
+            }
+
+            // Remove padding to get exact length
+            decodedLength = (int)((uint)length / 4 * 3) - paddingCount;
+            return true;
+        }
+
+        private static readonly IndexOfAnyValues<byte> validBase64Bytes = IndexOfAnyValues.Create(EncodingMap);
+
+        private static readonly IndexOfAnyValues<char> validBase64Chars = IndexOfAnyValues.Create("ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/");
```

Comment:
Per naming-convections (need to update the references too):
```suggestion
        private static readonly IndexOfAnyValues<byte> s_validBase64Bytes = IndexOfAnyValues.Create(EncodingMap);

        private static readonly IndexOfAnyValues<char> s_validBase64Chars = IndexOfAnyValues.Create("ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/");
```